src/main/java/bio/terra/workspace/service/spendprofile/SpendProfileService.java

```diff
@@ -0,0 +1,76 @@
+package bio.terra.workspace.service.spendprofile;
+
+import bio.terra.workspace.app.configuration.external.SpendProfileConfiguration;
+import bio.terra.workspace.common.utils.SamUtils;
+import bio.terra.workspace.service.iam.SamService;
+import bio.terra.workspace.service.spendprofile.exceptions.SpendUnauthorizedException;
+import com.google.common.collect.Maps;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * A service for retrieving and authorizing the use of {@link SpendProfile}s.
+ *
+ * <p>TODO: Integrate with the Spend Profile Manager component instead of doing our own in-memory
+ * configuration of spend profiles.
+ */
+@Component
+public class SpendProfileService {
+  private final Logger logger = LoggerFactory.getLogger(SpendProfileService.class);
+  private final SamService samService;
+  private final Map<SpendProfileId, SpendProfile> spendProfiles;
+
+  @Autowired
+  public SpendProfileService(
+      SamService samService, SpendProfileConfiguration spendProfileConfiguration) {
+    this(samService, parse(spendProfileConfiguration.getSpendProfiles()));
+  }
+
+  public SpendProfileService(SamService samService, List<SpendProfile> spendProfiles) {
+    this.samService = samService;
+    this.spendProfiles = Maps.uniqueIndex(spendProfiles, SpendProfile::id);
+  }
+
+  /**
+   * Authorize the user to link the Spend Profile. Returns the {@link SpendProfile} associated with
+   * the id if there is one and the user is authorized to link it. Otherwise, throws a {@link
+   * SpendUnauthorizedException}.
+   */
+  public SpendProfile authorizeLinking(SpendProfileId spendProfileId, String userAccessToken) {
```

Comment:
Should this take an `AuthenticatedUserRequest` here instead of the token directly? That's how other services have encapsulated permissions objects

Reply:
I'll buy the consistency argument, but other than JobService, none of them need anything outside of the user token. (I also don't know what it means to have an AuthenticatedUserRequest with an Optional.empty token, which is the one field we need here and everywhere else.) Going ahead with AuthenticatedUserRequest for this PR.